com.unity.render-pipelines.universal/Runtime/ShadowUtils.cs

```diff
@@ -168,9 +203,20 @@ public static Vector4 GetShadowBias(ref VisibleLight shadowLight, int shadowLigh
 
         public static void SetupShadowCasterConstantBuffer(CommandBuffer cmd, ref VisibleLight shadowLight, Vector4 shadowBias)
         {
-            Vector3 lightDirection = -shadowLight.localToWorldMatrix.GetColumn(2);
             cmd.SetGlobalVector("_ShadowBias", shadowBias);
-            cmd.SetGlobalVector("_LightDirection", new Vector4(lightDirection.x, lightDirection.y, lightDirection.z, 0.0f));
+
+            // Light direction is currently used in shadow caster pass to apply shadow normal offset (normal bias).
+            if(shadowLight.lightType == LightType.Directional)
+            {
+                Vector3 lightDirection = -shadowLight.localToWorldMatrix.GetColumn(2);
+                cmd.SetGlobalVector("_ShadowCastingLightParameters", new Vector4(lightDirection.x, lightDirection.y, lightDirection.z, 0.0f));
+            }
+            else
+            {
+                // For punctual lights, computing light direction at each vertex position provides more consistent results (shadow shape does not change when "rotating the point light" for example)
+                Vector3 lightPosition = shadowLight.localToWorldMatrix.GetColumn(3);
+                cmd.SetGlobalVector("_ShadowCastingLightParameters", new Vector4(lightPosition.x, lightPosition.y, lightPosition.z, 0.0f));
```

Comment:
You're right, it's better to avoid changing the meaning of constant variables.
Thanks for the suggestions, I pushed change 7a92566 where _LightPosition and _LightDirection are separate variables named more appropriately.